validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/BlockProductionDuty.java

```diff
@@ -96,6 +97,14 @@ public String getProducedType() {
 
   public SafeFuture<SignedBeaconBlock> signBlock(
       final ForkInfo forkInfo, final Optional<BeaconBlock> unsignedBlock) {
+    unsignedBlock.ifPresent(
+        beaconBlock ->
+            checkArgument(
+                beaconBlock.getSlot().equals(slot),
+                "Unsigned block slot ( "
+                    + beaconBlock.getSlot()
+                    + ") does not match expected slot "
+                    + slot));
```

Comment:
Can use same %s format trick as above to avoid creating the string until its actually needed.

I'd probably extract the `unsignedBlock.orElseThrow` below to a variable and then you can just use it directly here as well rather than needing an `ifPresent` block.